discord/bot.go

```diff
@@ -667,10 +675,22 @@ func (guild *GuildState) handleMessageCreate(s *discordgo.Session, m *discordgo.
 						}
 					}
 
+					guild.PrivateStateMsg.privateChannelID = privateChannelIDTemp;
+
 					guild.handleGameStartMessage(s, m, room, region, initialTracking)
+					guild.PrivateStateMsg.idUsernameMap = make(map[string]string)
```

Comment:
It's hard to say for certain (depending on the usage and typical use case), but these assignments could cause a race condition. If you have a `lock` in the PrivateStateMsg anyways, I would recommend making a constructor function that locks the state message, sets all these fields, and then unlocks. Again, not positive there is a race condition here, but there certainly could be. If you have a lock in the statemsg struct anyways, then I'd recommend using it here 👍

Reply:
When you say constructor function, do you mean a kind of factory method proposed by this website?: http://www.golangpatterns.info/object-oriented/constructors

Also, would I do something similar to this?:

`func (psm *PrivateStateMessage) Initialize(){
	psm.lock.Lock();
	defer psm.lock.Unlock()

	psm.idUsernameMap = make(map[string]string)
	psm.printedUsers = make([]string, 0)
	psm.currentUserID = ""
	psm.message = nil;
}`

Reply:
Yes, that looks perfect! I use the term "constructor" pretty loosely (thanks Java background), but yes, the basic idea is to define a single function that sets all the initial values for you. And it's particularly convenient here, because we can have the constructor handle the locking behavior, and not have to litter the other parts of the code with it (more error-prone if we forget to unlock, and more duplicated code)

Reply:
Sounds great! Thank you for your help! :D